docs/source/bundles.rst

```diff
@@ -0,0 +1,275 @@
+of ``$ZIPLINE_ROOT/data/<bundle>`` which is named with the current date. This
+makes it possible to look at older data or even run backtests with this older
+copy. This makers it easier to reproduce backtest results later.
+
+One drawback of saving all of this data by default is that the data directory
+may grow quite large even if you do not want to use the data. To solve this
+problem there is another command ``clean`` which will clear data bundles based
+on some time constraints.
+
+For example:
+
+.. code-block:: bash
+
+   # clean everything older than <date>
+   $ python -m zipline clean <bundle> --before <date>
+
+   # clean everything newer than <date>
+   $ python -m zipline clean <bundle> --after <date>
+
+   # keep everything in the range of [before, after] and delete the rest
+   $ python -m zipline clean <bundle> --before <date> --after <after>
+
+   # clean all but the last <int> runs
+   $ python -m zipline clean <bundle> --keep-last <int>
+
+
+Running Backtests with Data Bundles
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that the data has been ingested we can use it to run backtests with the
+``run`` command. This can be specified with the ``--bundle`` option like:
+
+.. code-block:: bash
+
+   $ python -m zipline run --bundle <bundle> --algofile algo.py ...
+
+
+We may also specify the date to use to look up the bundle data with the
+``--bundle-date`` option. This will cause us to the the most recent bundle
+ingestion that is less than or equal to the ``bundle-date``. This is how we can
+run backtests with older data. The reason that this uses a less than or equal to
+relationship is that we can specify the date that we ran an old backtest and get
+the same data that would have been available to us on that date. the
+``bundle-date`` defaults to the current day to use the most recent data.
+
+Default Data Bundles
+~~~~~~~~~~~~~~~~~~~~
+
+.. _quandl-data-bundle:
+
+Quandl WIKI Bundle
+``````````````````
+
+By default zipline comes with the ``quandl`` data bundle which uses quandl's
+`WIKI dataset <https://www.quandl.com/data/WIKI>`_. This bundle includes daily
+pricing data, splits, cash dividends, and asset metadata. This is the bundle
+that ``run`` will use by default if no other bundle is specified. To ingest this
+data bundle we recommend creating an account on quandl.com to get an API key to
+be able to make more API requests. Once we have an API key we may run:
+
+.. code-block:: bash
+
+   $ QUANDL_API_KEY=<api-key> python -m zipline ingest quandl
+
+though we may still run ``ingest`` as an anonymous quandl user (with no API
+key). We may also set the ``QUANDL_DOWNLOAD_ATTEMPTS`` environment variable to
+an integer which is the number of attempts that should be made to download data
+from quandls servers. By default this will be 5, meaning that we will retry each
+attempt 5 times.
+
+.. note::
+
+   This is not the total number of allowed failures, just the number of allowed
```

Comment:
It's unclear what the word "this" refers to here.  There are several other usages of the word "This" in this paragraph that should probably be replaced with explicit nouns.  (e.g. "this bundle includes" vs "the quandl bundle includes.).